memiavl/db.go

```diff
@@ -341,6 +354,19 @@ func (db *DB) checkBackgroundSnapshotRewrite() error {
 			return fmt.Errorf("background snapshot rewriting failed: %w", result.err)
 		}
 
+		// wait for potential pending wal writings to finish, to make sure we catch up to latest state.
+		// in real world, block execution should be slower than wal writing, so this should not block for long.
+		for {
+			committedVersion, err := db.CommittedVersion()
+			if err != nil {
+				return fmt.Errorf("get wal version failed: %w", err)
+			}
+			if db.lastCommitInfo.Version == committedVersion {
+				break
+			}
+			time.Sleep(time.Nanosecond)
+		}
```

Comment:
sounds good, sth like
```suggestion
		success := false
		for i := 0; i < int(10*time.Second/time.Nanosecond); i++ {
			committedVersion, err := db.CommittedVersion()
			if err != nil {
				return fmt.Errorf("get wal version failed: %w", err)
			}
			if db.lastCommitInfo.Version == committedVersion {
				success = true
				break
			}
			time.Sleep(time.Nanosecond)
		}
		if !success {
			return errors.New("get wal version timeout")
		}
```

Reply:
the exact duration for timeout is hard to decide, it depends on the buffer size, and file writing speed.